hera_mc/cm_utils.py

```diff
@@ -230,6 +233,19 @@ def listify(X):
     return [X]
 
 
+def match_list(a_obj, b_obj, upper=False):
+    if not isinstance(a_obj, list):
+        a_obj = [a_obj]
```

Comment:
Do you only want to check for lists? If an array is passed then this will make a 1-element list that contains the array in that one element.

Reply:
For this and comment below I made quite a few changes that occurred while I was writing the docstrings plus added pytesting for new stuff in cm_utils